File: src/lessons/welcome/methods/picture/PatternPictureEntity.c
//RemoteBuggle

void bigSquare();
void squareB(Color c);
void squareA(Color c);
void mark() ;


/* BEGIN TEMPLATE */

void run(){
	/* BEGIN SOLUTION */
	bigSquare();
	forward(4);
	bigSquare();

	backward(4);
	left();
	forward(4);
	right();

	bigSquare();
	forward(4);
	bigSquare();

}
void mark() {
	brushDown();
	brushUp();
}

void squareA(Color c) {
	setBrushColor(c);

	forward(1);
	mark();

	left();
	forward(1);

	left();
	forward(1);
	mark();

	left();
	forward(1);
	left();
}

void squareB(Color c) {
	setBrushColor(c);
	mark();

	forward(1);

	left();
	forward(1);
	mark();

	left();
	forward(1);

	left();
	forward(1);
	left();
}

void bigSquare() {
	squareA(red);
	forward(2);
	squareB(blue);
	backward(2);
	left();
	forward(2);
	right();
	squareB(yellow);
	forward(2);
	squareA(green);

	backward(2);
	left();
	backward(2);
	right();
/* END SOLUTION */
}

/* END TEMPLATE */

File: src/lessons/welcome/methods/picture/PatternPictureEntity.java
package lessons.welcome.methods.picture;

import java.awt.Color;
import plm.universe.bugglequest.SimpleBuggle;

public class PatternPictureEntity extends SimpleBuggle {

  /* BEGIN TEMPLATE */
  public void run()
  {
    /* BEGIN SOLUTION */
    bigSquare();
    forward(4);
    bigSquare();

    backward(4);
    left();
    forward(4);
    right();

    bigSquare();
    forward(4);
    bigSquare();
  }
  void mark()
  {
    brushDown();
    brushUp();
  }

  void squareA(Color c)
  {
    setBrushColor(c);

    stepForward();
    mark();

    left();
    stepForward();

    left();
    stepForward();
    mark();

    left();
    stepForward();
    left();
  }

  void squareB(Color c)
  {
    setBrushColor(c);
    mark();

    stepForward();

    left();
    stepForward();
    mark();

    left();
    stepForward();

    left();
    stepForward();
    left();
  }

  void bigSquare()
  {
    squareA(Color.RED);
    forward(2);
    squareB(Color.BLUE);
    backward(2);
    left();
    forward(2);
    right();
    squareB(Color.YELLOW);
    forward(2);
    squareA(Color.GREEN);

    backward(2);
    left();
    backward(2);
    right();
    /* END SOLUTION */
  }
  /* END TEMPLATE */
}
